### PR TITLE
BTreeMap: correct node size test case for choices of B

### PR DESCRIPTION
**Pull request by ssomers**
r? @Mark-Simulacrum

### CODE REVIEW

File: library/alloc/src/collections/btree/node/tests.rs [hunk 1]
```diff
@@ -114,6 +114,5 @@ fn test_partial_cmp_eq() {
 fn test_sizes() {
     assert_eq!(core::mem::size_of::<LeafNode<(), ()>>(), 16);
     assert_eq!(core::mem::size_of::<LeafNode<i64, i64>>(), 16 + CAPACITY * 8 * 2);
-    assert_eq!(core::mem::size_of::<InternalNode<(), ()>>(), 112);
-    assert_eq!(core::mem::size_of::<InternalNode<i64, i64>>(), 112 + CAPACITY * 8 * 2);
+    assert_eq!(core::mem::size_of::<InternalNode<(), ()>>(), 16 + (CAPACITY + 1) * 8);
```
Comment (Mark-Simulacrum) [inline library/alloc/src/collections/btree/node/tests.rs:117]: Why did we remove the additional assert here?

Comment (ssomers) [reply library/alloc/src/collections/btree/node/tests.rs:117]: It seems to have very little chance of detecting anything missed by the 2nd and 3rd assert, for a complicated formula, but I'll add it again.

**Comment by ssomers** [comment]
Also stripped benchmarks that cause a stack overflow for bigger choices of B. There's little point in assessing the performance of 0.5 KB key-value pairs when the physical limit is below 3 KB.
@rustbot modify labels: -S-waiting-on-author +S-waiting-on-review

**Comment by Mark-Simulacrum** [comment]
@bors r+

box vs Box::new is interesting. I would not be opposed to using box fwiw, I think, but I expect that the difference is likely noise.

**Comment by ssomers** [comment]
> box vs Box::new is interesting.

Well I turned it into #81444.



> 
> 
> @bors r+
> 
> box vs Box::new is interesting. I would not be opposed to using box fwiw, I think, but I expect that the difference is likely noise.

**Pull request merged**